derive(Pwrite) doesn't compile for structs with non-Copy fields

I ran into this while trying to test something else, but it's definitely going to bite me in the future. Here's a simple example:
```rust
#[derive(Pwrite)]
struct A {
    pub y: u64,
}

#[derive(Pwrite)]
struct B {
    pub a: A,
}
```

This code fails to compile with this error:
```
error[E0507]: cannot move out of borrowed content
  --> src/main.rs:12:10
   |
12 | #[derive(Pwrite)]
   |          ^^^^^^ cannot move out of borrowed content
```

The expanded impls for `B` look like:
```rust
impl <'a> ::scroll::ctx::TryIntoCtx<::scroll::Endian> for &'a B {
    type
    Error
    =
    ::scroll::Error;
    type
    Size
    =
    usize;
    #[inline]
    fn try_into_ctx(self, dst: &mut [u8], ctx: ::scroll::Endian)
     -> ::scroll::export::result::Result<Self::Size, Self::Error> {
        use ::scroll::Pwrite;
        let offset = &mut 0;
        dst.gwrite_with(self.a, offset, ctx)?;
        Ok(*offset)
    }
}
impl ::scroll::ctx::TryIntoCtx<::scroll::Endian> for B {
    type
    Error
    =
    ::scroll::Error;
    type
    Size
    =
    usize;
    #[inline]
    fn try_into_ctx(self, dst: &mut [u8], ctx: ::scroll::Endian)
     -> ::scroll::export::result::Result<Self::Size, Self::Error> {
        (&self).try_into_ctx(dst, ctx)
    }
}
```

Removing the `#[derive(Pwrite)]` and pasting that code inline gives a more precise error:
```
error[E0507]: cannot move out of borrowed content
  --> src/main.rs:30:25
   |
30 |         dst.gwrite_with(self.a, offset, ctx)?;
   |                         ^^^^ cannot move out of borrowed content
```

Changing that line to borrow instead fixes this: `dst.gwrite_with(&self.a, offset, ctx)?;`. I haven't checked to see if that has a negative impact on the generated code, but since the `TryIntoCtx` impl for `T` already forwards to `&T` I can't imagine it'd make a difference.

Unfortunately changing the proc macro implementations to use `&self.#ident` fails because we don't have implementations of `TryIntoCtx` for references-to-primitives:
```
error[E0277]: the trait bound `&u32: scroll::ctx::TryIntoCtx<_>` is not satisfied
 --> tests/tests.rs:5:35
  |
5 | #[derive(Debug, PartialEq, Pread, Pwrite)]
  |                                   ^^^^^^ the trait `scroll::ctx::TryIntoCtx<_>` is not implemented for `&u32`
  |
  = help: the following implementations were found:
            <u32 as scroll::ctx::TryIntoCtx<scroll::Endian>>
```

This probably isn't hard to fix, though.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -79,13 +79,13 @@ fn impl_try_into_ctx(name: &syn::Ident, fields: &syn::FieldsNamed) -> proc_macro
             &syn::Type::Array(_) => {
                 quote! {
                     for i in 0..self.#ident.len() {
-                        dst.gwrite_with(self.#ident[i], offset, ctx)?;
+                        dst.gwrite_with(&self.#ident[i], offset, ctx)?;
                     }
                 }
             },
             _ => {
                 quote! {
-                    dst.gwrite_with(self.#ident, offset, ctx)?
+                    dst.gwrite_with(&self.#ident, offset, ctx)?
                 }
             }
         }

--- a/tests/tests.rs
+++ b/tests/tests.rs
@@ -134,30 +134,27 @@ fn test_array_copy (){
     assert_eq!(name, "hell");
 }
 
-#[derive(Pread, SizeWith)]
+#[derive(Debug, PartialEq, Eq, Pread, Pwrite, SizeWith)]
 struct Data7A {
     pub y: u64,
     pub x: u32,
 }
 
-#[derive(Pread, SizeWith)]
+#[derive(Debug, PartialEq, Eq, Pread, Pwrite, SizeWith)]
 struct Data7B {
     pub a: Data7A,
 }
 
 #[test]
 fn test_nested_struct() {
-    const BYTES: [u8; 12] = [
-        // y: u64
-        1, 0, 0, 0, 0, 0, 0, 0,
-        // x: u32
-        2, 0, 0, 0,
-    ];
+    let b = Data7B { a: Data7A { y: 1, x: 2 } };
     let size = Data7B::size_with(&LE);
     assert_eq!(size, 12);
+    let mut bytes = vec![0; size];
+    let written = bytes.pwrite_with(&b, 0, LE).unwrap();
+    assert_eq!(written, size);
     let mut read = 0;
-    let b: Data7B = BYTES.gread_with(&mut read, LE).unwrap();
-    assert_eq!(b.a.y, 1);
-    assert_eq!(b.a.x, 2);
+    let b2: Data7B = bytes.gread_with(&mut read, LE).unwrap();
     assert_eq!(read, size);
+    assert_eq!(b, b2);
 }

